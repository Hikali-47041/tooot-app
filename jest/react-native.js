jest.mock('react-native/Libraries/LayoutAnimation/LayoutAnimation', () => ({
  ...require.requireActual(
    'react-native/Libraries/LayoutAnimation/LayoutAnimation'
  ),
  configureNext: jest.fn()
}))
