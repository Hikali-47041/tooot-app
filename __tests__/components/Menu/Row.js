import React from 'react'
import { toBeDisabled } from '@testing-library/jest-native'
import { cleanup, fireEvent, render } from '@testing-library/react-native'

import MenuRow from '@components/Menu/Row'

expect.extend({ toBeDisabled })

describe('Testing component menu row', () => {
  afterEach(cleanup)

  it('with title only', () => {
    const { getByText, toJSON } = render(<MenuRow title='test title' />)
    getByText('test title')
    expect(toJSON()).toMatchSnapshot()
  })

  it('with title and content', () => {
    const { getByText, toJSON } = render(
      <MenuRow title='test title' content='test content' />
    )
    getByText('test title')
    getByText('test content')
    expect(toJSON()).toMatchSnapshot()
  })

  it('on press event', () => {
    const onPress = jest.fn()
    const { getByTestId, toJSON } = render(
      <MenuRow title='test' onPress={onPress} />
    )
    fireEvent.press(getByTestId('base'))

    expect(onPress).toHaveBeenCalled()
    expect(onPress).toHaveBeenCalledTimes(1)
    expect(toJSON()).toMatchSnapshot()
  })

  it('loading state', () => {
    const onPress = jest.fn()
    const { getByTestId, toJSON } = render(
      <MenuRow title='test' loading onPress={onPress} />
    )
    fireEvent.press(getByTestId('base'))

    expect(onPress).toHaveBeenCalledTimes(0)
    expect(getByTestId('base')).toBeDisabled()
    expect(toJSON()).toMatchSnapshot()
  })
})
