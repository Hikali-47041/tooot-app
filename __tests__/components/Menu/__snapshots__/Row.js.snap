// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component menu row loading state 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
    }
  }
  testID="base"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexBasis": "70%",
          "flexDirection": "row",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "flex": 1,
              "fontSize": 16,
            },
            Object {
              "color": "rgb(18, 18, 18)",
            },
          ]
        }
      >
        test
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing component menu row on press event 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
    }
  }
  testID="base"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexBasis": "70%",
          "flexDirection": "row",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "flex": 1,
              "fontSize": 16,
            },
            Object {
              "color": "rgb(18, 18, 18)",
            },
          ]
        }
      >
        test
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing component menu row with title and content 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
    }
  }
  testID="base"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexBasis": "70%",
          "flexDirection": "row",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "flex": 1,
              "fontSize": 16,
            },
            Object {
              "color": "rgb(18, 18, 18)",
            },
          ]
        }
      >
        test title
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexBasis": "30%",
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 16,
            },
            Object {
              "color": "rgb(135, 135, 135)",
              "opacity": 1,
            },
          ]
        }
      >
        test content
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing component menu row with title only 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
    }
  }
  testID="base"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexBasis": "70%",
          "flexDirection": "row",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "flex": 1,
              "fontSize": 16,
            },
            Object {
              "color": "rgb(18, 18, 18)",
            },
          ]
        }
      >
        test title
      </Text>
    </View>
  </View>
</View>
`;
